sql/report_util_functions.sql

```diff
@@ -723,6 +723,31 @@ BEGIN
 END$$
 DELIMITER ;
 
+-- patientReasonForConsultationIsUnplannedAid
+
+DROP FUNCTION IF EXISTS patientReasonForConsultationIsUnplannedAid;
+
+DELIMITER $$
+CREATE FUNCTION patientReasonForConsultationIsUnplannedAid(
+    p_patientId INT(11)) RETURNS TINYINT(1)
+    DETERMINISTIC
+BEGIN 
+    DECLARE patientIsUnplannedAid TINYINT(1) DEFAULT 0;
+
+    DECLARE uuidPatientIsUnplannedAid VARCHAR(38) DEFAULT "17a3b24b-e107-49fe-8b0d-69c3b7e60f4c";
+
+    SELECT TRUE INTO patientIsUnplannedAid
+    FROM person p
+    JOIN patient_program pp ON pp.patient_id = p.person_id AND pp.voided = 0 AND p_patientId = pp.patient_id
+    JOIN patient_program_attribute ppt ON ppt.patient_program_id = pp.patient_program_id
+    JOIN concept c ON c.concept_id = ppt.value_reference
+    WHERE  p.voided = 0
+        AND c.uuid = uuidPatientIsUnplannedAid;
```

Comment:
@IipperMangesh Please add a `LIMIT 1` at the end of the query to prevent the 'too many records' error from occurring